File: LeetCode/01/lc121.py
# lc121.py

# LeetCode 121 Best Time to Buy and Sell Stock `E`
# A~0c03
# acc | 99%

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice, maxProfit = 65536, 0
        for i in prices:
            if i < minPrice:
                minPrice = i
            elif i - minPrice > maxProfit:
                maxProfit = i - minPrice
        return maxProfit

File: LeetCode/01/lc121.c
/* lc121.c */

/* LeetCode 121 Best Time to Buy and Sell Stock `E` */
/* A~0c03 */
/* acc | 99% | 23' */

int maxProfit(int *prices, int pricesSize)
{
    int minPrice = 65536, maxProfit = 0;
    for (int i = 0; i < pricesSize; i++)
        if (prices[i] < minPrice)
            minPrice = prices[i];
        else if (prices[i] - minPrice > maxProfit)
            maxProfit = prices[i] - minPrice;

    return maxProfit;
}
